### ISSUE TITLE
Future incompatibility of `fs_extra v1.2.0` and `wasmparser v0.78.2`

### ISSUE THREAD

**Issue by jakmeier** [near/nearcore#8723]
With Rust 1.68.0 (which we use since #8721) there is a future compatibility warning.

warning: the following packages contain code that will be rejected by a future version of Rust: fs_extra v1.2.0, wasmer-singlepass-backend-near v0.18.1, wasmparser v0.78.2

This is the relevant change: https://github.com/rust-lang/rust/pull/103418

I think `fs_extra` we could update without too much trouble.

But for `wasmparser`, I think we want the exact version to be sure we have no behavioral changes for older protocol version. I guess maybe we want to ask for a 0.78.3?

**Comment by nagisa** [comment]
I don't think it is critical that we act on this right away – future compatibility lints tend to stick around as warnings for quite a while before they become an error.

Knowing `wasmparser`'s approach to maintenance, I strongly doubt they'll be onboard with releasing patch versions for past releases. I don't think it is reasonable for us to expect them to, either. After all, if there were 100 projects using random, locked versions of wasmparser, requests for patch versions like these would be a never-ending deluge. Not to mention, that were there another compatibility lint in the future, maintainers would have to suffer through an encore such an effort.

Two options I see that would fix this properly:

1. Vendor the affected libraries and adjust;
2. (Proper) limited replayability.
